commitment/taproot.go

```diff
@@ -113,6 +115,56 @@ func NewPreimageFromBranch(branch txscript.TapBranch) *TapscriptPreimage {
 	}
 }
 
+// TapTreeToSibling constucts a taproot sibling hash from Tapscript tree nodes,
+// to be used with a TapCommitment tree root to derive a tapscript root. This
+// could be multiple TapLeaf objects, or a representation of a TapBranch.
+func NewPreimageFromTapscriptTreeNodes(
+	tn asset.TapscriptTreeNodes) (*TapscriptPreimage, error) {
+
+	var preimage *TapscriptPreimage
+
+	maybeLeaves := tn.GetLeaves().UnwrapToPtr()
+	if maybeLeaves != nil {
```

Comment:
nit: `WhenSome` instead?

Reply:
Would be same indentation fwiw, but can also chalk it up to a personal style thing. As this wants to return an error, `MapOptionZ` can be used as well.